docker_engine_api/request_builder.ex

```diff
@@ -152,46 +152,27 @@ defmodule DockerEngineAPI.RequestBuilder do
   @type response_mapping :: [{status_code, false | %{} | module()}]
 
   @doc """
-  Evaluate the response from a Tesla request.
-  Decode the response for a Tesla request.
+  Evaluate and decode the response from a Tesla request.
 
-  ### Parameters
-
-  - `result` (Tesla.Env.result()): The response from Tesla.request/2.
-  - `mapping` ([{http_status, struct}]): The mapping for status to struct for decoding.
-
-  ### Returns
-
-  - `{:ok, struct}` or `{:ok, Tesla.Env.t()}` on success
-  - `{:error, term}` on failure
+  - `result` (Tesla.Env.result()): The Tesla response.
+  - `mapping` ([{http_status, struct}]): Status-to-struct mapping for decoding.
   """
-  @spec evaluate_response(Tesla.Env.result(), response_mapping) :: {:ok, struct()} | Tesla.Env.result()
+  @spec evaluate_response(Tesla.Env.result(), response_mapping) :: {:ok, struct()} | {:ok, Tesla.Env.t} | {:error, Tesla.Env.t}
   def evaluate_response({:ok, %Tesla.Env{} = env}, mapping) do
-    resolve_mapping(env, mapping, nil)
-  end
-
-  def evaluate_response({:error, _} = error, _), do: error
-
-  defp resolve_mapping(%Tesla.Env{status: status} = env, [{mapping_status, struct} | _], _)
-      when status == mapping_status do
-    decode(env, struct)
+    status = env.status
+    mapping
+    |> Enum.find_value({:error, env}, fn
+      {^status, struct} -> decode(env, struct)
+      _ -> {:error, env}
```

Comment:
Ah should propably return mil here so the Find function falls back to its default